bluesky/callbacks/core.py

```diff
@@ -3,15 +3,65 @@
 """
 from itertools import count
 import warnings
+import os
+
 from collections import deque, namedtuple, OrderedDict
 import time as ttime
+from functools import wraps as _wraps, partial as _partial
 
 from datetime import datetime
 import logging
 from ..utils import ensure_uid
+from event_model import DocumentNames
+
+
+logger = logging.getLogger(__name__)
+
+
+def make_callback_safe(func=None, *, logger=None):
+    if func is None:
+        return _partial(make_callback_safe, logger=logger)
+
+    @_wraps(func)
+    def inner(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception:
+            debug_mode = os.environ.get("BLUESKY_DEBUG_CALLBACKS", False)
+            if logger is not None:
+                if debug_mode:
+                    msg = f"Exception in {func}"
+                else:
+                    msg = (
+                        f"An exception raised in the callback {func} "
+                        + "is being suppressed to not interupt plan "
+                        + "execution.  To investigate try setting the "
+                        + "BLUESKY_DEBUG_CALLBACKS env to '1'"
```

Comment:
...or really to _anything_.  Should we make it so the `BLUESKY_DEBUG_CALLBACKS=0` turn has no effect?

Reply:
@mrakitin What do you think? Is it better to have this switch on set/unset (ignoring the value) or to provide a way to explicitly set this to off?

Reply:
I think the first way (a switch) is better, as one does not have to worry about setting it to anything by default, and resetting `BLUESKY_DEBUG_CALLBACKS= <shell command>` will work if needed. Same was as https://docs.python.org/3/using/cmdline.html#envvar-PYTHONUNBUFFERED works (I saw people use `1` or `x` to enable it).

Reply:
Also, it seems we don't need `+`s here.